sdk/spring-sdk/src/main/scala/kalix/springsdk/impl/ComponentMethod.scala

```diff
@@ -17,13 +17,47 @@
 package kalix.springsdk.impl
 
 import java.lang.reflect.Method
-
 import com.google.protobuf.Descriptors
 import kalix.springsdk.impl.reflection.ParameterExtractor
+import kalix.springsdk.impl.reflection.ParameterExtractors.AnyBodyExtractor
 
 // Might need to have one of each of these for unary, streamed out, streamed in and streamed.
+/**
+ * @param method
+ *   method from the annotated class
+ * @param grpcMethodName
+ *   'rpc' method name. When it has a methodsMap - see below - the name is not the same as the method above but a
+ *   synthetic one.
+ * @param parameterExtractors
+ *   To extract the value of a parameter from protobuf.Any
+ * @param requestMessageDescriptor
+ * @param inputClass2Method
+ *   This component method may represent multiple methods in a class. For example, if a class is annotated with the same
+ * @Subscription.Topic("xyz")
+ *   in two different methods then this methodsMap keeps reference to those two methods.
+ */
 case class ComponentMethod(
     method: Option[Method],
     grpcMethodName: String,
     parameterExtractors: Array[ParameterExtractor[InvocationContext, AnyRef]],
-    requestMessageDescriptor: Descriptors.Descriptor)
+    requestMessageDescriptor: Descriptors.Descriptor,
+    inputClass2Method: Map[String, Method] = Map()) {
+
+  def lookupMethod(inputTypeUrl: String): JavaMethod = {
+    if (inputClass2Method.isEmpty) {
+      JavaMethod(method, parameterExtractors)
+    } else {
+      val method = inputClass2Method.get(inputTypeUrl)
+      val methodParameterTypes = method.get.getParameterTypes();
+      val extractors = parameterExtractors.collect {
+        //FIXME is it safe to pick the last parameter. An action has one and View has two. Is it in the View always the second the event? are we enforcing that?
+        case extractor @ AnyBodyExtractor(cls)
+            if cls.getName.equals(methodParameterTypes(methodParameterTypes.size - 1).getName) =>
+          extractor.asInstanceOf[ParameterExtractor[InvocationContext, AnyRef]]
+      }
+      JavaMethod(method, extractors)
```

Comment:
From what I understand we won't be able to link it to the `typeUrl` as we generate the JavaMethod prior to starting the application. We don't get the `typeUrl` until the application is up and running.

AFAIK we have two facets of the springSDK. One that loads/generates all the .protos, routers and providers. And a second that interacts with the user and the proxy. This division makes me reformat the code so this is explicit. Maybe grouping things in two different packages. `load` and `server`, for example. What do you think @octonato

Reply:
You are right, sorry. That was not a problem. The problem is that if we try to build the `JavaMethod`s directly into the `typeUrl2Methods` we need there the `parameterExtractors` but these are not available until later. We create the `typeUrl2Methods` in the Factories - `ActionDescriptorFactor` and so on - while the `parametersExtractors` are created/extracted in the `ComponentDescriptor` factory method. 

It seems I don't want this change to happen :|

Reply:
The parameters extractors are dependent on the type of ServiceMethod. The reason why it's in the ComponentDescriptor is because we just add the logic there. 

If we move the logic to each impl of ServiceMethod, then we don't have that problem anymore.